lib/queue/queue.py

```diff
@@ -98,27 +118,29 @@ def get_or_create_queues(self, queue_name: str) -> List[boto3.resources.base.Ser
             else:
                 raise
 
-    def get_sqs(self) -> boto3.resources.base.ServiceResource:
+    def create_queue(self, queue_name: str):
         """
-        Get an instantiated SQS - if local, use local alternative via elasticmq
+        Create a queue by name.
         """
-        deploy_env = get_environment_setting("DEPLOY_ENV")
-        if deploy_env == "local":
-            logger.info(f"Using ElasticMQ Interface")
-            with self.sqs_lock:
-                return boto3.resource('sqs',
-                                      region_name=(get_environment_setting("AWS_DEFAULT_REGION") or 'eu-central-1'),
-                                      endpoint_url=(get_environment_setting("ELASTICMQ_URI") or 'http://presto-elasticmq:9324'),
-                                      aws_access_key_id=(get_environment_setting("AWS_ACCESS_KEY_ID") or 'x'),
-                                      aws_secret_access_key=(get_environment_setting("AWS_SECRET_ACCESS_KEY") or 'x'))
-        else:
-            logger.info(f"Using SQS Interface")
-            with self.sqs_lock:
-                return boto3.resource('sqs', region_name=get_environment_setting("AWS_DEFAULT_REGION"))
+        attributes = {}
+        if queue_name.endswith('.fifo'):
+            attributes['FifoQueue'] = 'true'
+            attributes['ContentBasedDeduplication'] = 'true'
+        return self.get_sqs().create_queue(QueueName=queue_name, Attributes=attributes)
+
+    def send_message(self, queue_name: str, message: schemas.Message):
+        """
+        Send a message to a specific queue.
+        """
+        queue = self.get_or_create_queue(queue_name)
```

Comment:
Do we really need to look up the queue each time we send a message? I would have hoped we could store a reference after looking it up once

Reply:
My understanding is that if we _do_ store it we introduce a bunch of state which causes our problem in the first place

Reply:
The first call it will create the resource object, but then it will return a thread local reference on every subsequent request (within that thread). I think this is necessary due to the context switching that could happen otherwise...